astro/src/content/quickstarts/quickstart-javascript-express-web.mdx

```diff
@@ -0,0 +1,320 @@
+  lang="ts" />
+
+#### Run the App!
+You should now be able to start your Express application with 
+
+```shell
+npm run dev
+```
+
+### Walkthrough Example
+
+Make sure once again that you are in the directory `complete-application/src`.
+Create a file called `index.ts` and start by setting up your imports and constants. These constants have already been defined by running the kickstart and `setup.js`.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="top"/>
+
+<Aside type="note">
+If you want to see where the FusionAuth values came from, they can be found in the 
+FusionAuth app (http://localhost:9011/admin). The tenant ID is found on the Tenants
+page. To see the client ID and client secret, go to the Applications page and 
+click the `View` icon under the actions for the Changebank application. You'll 
+find the client id and client secret values in the `OAuth configuration` section.
+</Aside>
+
+#### Static files
+
+We need a way to serve our static files, luckily Express makes this easy with `express.static`. These files will include all of the CSS and images that are required for the UI of the ChangeBank application.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="static"/>
+
+#### Homepage
+
+In order to show off our homepage we need to create a route with the base url `/`. In this route check to see if `userDetails` cookie has any user values and correctly respond with either the `home.html` or `account.html` template. Also if there is no user you can generate the PKCE challenge pair and store them into `userSession` cookie (take note this is an http-only cookie).
+
+<Aside type="note">
+In this file we will use the package [pkce-challenge](https://www.npmjs.com/package/pkce-challenge) to generate a Proof Key for Code Exchange (PKCE) challenge pair. Read more about [PKCE](https://www.oauth.com/oauth2-servers/pkce/authorization-request/).
+</Aside>
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="homepage"/>
+
+#### Login Route
+
+In the Changebank application there is a `Login` button, when clicked will open the `/login` route (you can find this in `complete-application/templates/home.html`). Have this redirect them to FusionAuth so that FusionAuth can present them with a login page.
+
+`index.ts`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/src/index.ts"}
+  lang="ts"
+  tags="login"/>
+
+You can always find this url if you view the details of your Application within FusionAuth, under "OAuth2 & OpenID Connect Integration details" you will find the property `OAuth IdP login URL`. The endpoint that you will hit on FusionAuth is `/oauth2/authorize` below you can find a list of the needed parameters for FusionAuth's [OAuth Endpoints](https://fusionauth.io/docs/v1/tech/oauth/endpoints)
```

Comment:
I think we should rephrase this paragraph, it's kind of confusing

Reply:
How about

The OAuth IdP login URL can be found in the "OAuth2 & OpenID Connect Integration details" section of your application's settings in FusionAuth. The endpoint that you will hit on FusionAuth is /oauth2/authorize. You can find a list of the required parameters for FusionAuth's OAuth endpoints here: [OAuth Endpoints](/docs/v1/tech/oauth/endpoints)

Reply:
Awesome!